miniapp/USERGUIDE.md

```diff
@@ -622,6 +622,44 @@ mini app scheme and should close external webview.
 
 Using `#ExternalResultHandler.emitResult(String)` to transmit the url string to mini app view.
 
+### File picking
+**API Docs:** [MiniAppFilePicker](api/com.rakuten.tech.mobile.miniapp.file/-mini-app-file-picker/)
+
+The mini app is able to pick a file which is requested using [input type="file"](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file).
+HostApp needs to use `MiniAppFilePicker` object to pass through `MiniApp.create(appId: String, miniAppMessageBridge: MiniAppMessageBridge, miniAppFilePicker: MiniAppFilePicker)`.
+
+- At first, HostApp should implement `MiniAppFilePicker` in the `Activity`.
+
+```kotlin
+var filePathHostCallback: ValueCallback<Array<Uri>>? = null
+val filePickingReqCode = REQUEST_CODE // define a request code in HostApp
+
+miniAppFilePicker = object : MiniAppFilePicker {
+    override fun requestFile(
+        filePathCallback: ValueCallback<Array<Uri>>?,
+        callback: (requestCode: Int) -> Unit
+    ) {
+        filePathHostCallback = filePathCallback
+        callback.invoke(filePickingReqCode) // invoke the request code what is defined by HostApp.
+    }
+}
```

Comment:
I think the implementation could be simplified for the Host App: it doesn't seem like it's necessary to save the `filePathCallback` within the Host App. This could be contained within `MiniAppFilePicker`. For example, something like this:

```
val filePickingReqCode = REQUEST_CODE 
val miniAppFilePicker = MiniAppFilePicker(requestCode = filePickingReqCode)

override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
    super.onActivityResult(requestCode, resultCode, data)
    if (requestCode == filePickingReqCode && resultCode == Activity.RESULT_OK) {
        data?.let { intent ->
            val result: Uri? = intent.data
            miniAppFilePicker.onReceivedFile(arrayOf(result) ?: emptyArray())
        }
    }
}
```

Reply:
Thanks, I have updated this part.